### ISSUE TITLE
Impossible to launch app on android

### ISSUE THREAD

**Issue by Seabhacsuil** [beeware/briefcase#561]
**Describe the bug**
App crashes when launched on Android after displaying the "Made with Beeware" screen.

**To Reproduce**
1. Follow tutorial
2. Modify with own code
3. create > build > run android

**Expected behavior**
Get on my phone what I got on my computer

**Environment:**
 - Operating System: Windows 10 / Android 11 with One UI 3.0
 - Python version: 3.8.7
 - Software versions:
   - Briefcase: 0.3.4
   - Toga: 0.3.0

**Additional context**
I followed the tutorial, only replacing the Hello World with my own code, using buttons and labels. Everything works fine on computer. I tried also on an android emulator as written in the tutorial but the emulator cannot be launched.

**Comment by MERP-Solutions** [comment]
I have the exact same problem.
Briefcase 0.3.4 and Toga 0.3.0
Windows 10 / Android 11
Python : 3.8.7

The program works on windows from scratch till packaging. It installs on android but then it keeps force closing.
Error in android is :
```
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3388)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3527)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2123)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:107)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:214)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7710)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:516)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
02-01 13:38:00.002 29775 29775 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void org.beeware.android.IPythonApp.onCreate()' on a null object reference
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at org.beeware.android.MainActivity.onCreate(MainActivity.java:247)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:7820)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:7809)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1318)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3363)
02-01 13:38:00.002 29775 29775 E AndroidRuntime:        ... 11 more
02-01 13:38:17.421 30094 30094 E AndroidRuntime: FATAL EXCEPTION: main
02-01 13:38:17.421 30094 30094 E AndroidRuntime: Process: nl.teamworks.twsticker, PID: 30094
02-01 13:38:17.421 30094 30094 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{nl.teamworks.twsticker/org.beeware.android.MainActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void org.beeware.android.IPythonApp.onCreate()' on a null object reference
```

**Comment by KabirAcharya** [comment]
I'm unable to reproduce this error on Python 3.9. This may be version specific.

**Comment by HKLM** [comment]
I too had this issue. I was able to figure out why it would not work on android by using Android Studio and reviewing the output in the Logcat tab. 
I have noticed that briefcase `briefcase update android -d` does not seem to actually update ALL the required dependencies.  I did correctly add the packages to the required section of my projects "pyproject.toml" file.  To fix it-manually copy the needed package folders with its contents, that exist in the "site-packages" folder, but not in the android "app_packages" folder.
`~\beeware-venv\Lib\site-packages\` 
copy missing packages to
`~\<app package name>\android\gradle\<app title>\app\src\main\assets\python\app_packages`
then a android build and run and it then works.

Briefcase 0.3.4 release
Windows 11 x64 / Android API 28, 29, 30
Python : 3.8.7 64bit
Android Stuido 4.1.2
